javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/declarations/JavaParserSymbolDeclaration.java

```diff
@@ -99,95 +78,8 @@ public static int getParamPos(Node node) {
         throw new IllegalArgumentException();
     }
 
-    @Override
-    public String toString() {
-        return "JavaParserSymbolDeclaration{" +
-                "name='" + name + '\'' +
-                ", wrappedNode=" + wrappedNode +
-                '}';
-    }
-
-    @Override
-    public String getName() {
-        return name;
-    }
-
-    @Override
-    public boolean isField() {
-        return false;
-    }
-
-    @Override
-    public boolean isParameter() {
-        return false;
-    }
-
-    @Override
-    public boolean isPattern() {
-//        return getWrappedNode() instanceof PatternExpr;
-        return false;
-    }
-
-    @Override
-    public boolean isType() {
-        return false;
+    private JavaParserSymbolDeclaration() {
+        // This private constructor is used to hide the public one
```

Comment:
Removing it immediately makes me a little nervous, but I'm not completely opposed to this.

In either case, `@deprecated` annotations and documentation need to be added to the class / constructor so that users of this class can be guided to what they _should_ be using instead of this.

Reply:
Do you think this can be a issue? I was thinking it could be safe to remove, since it already had a private constructor and the only way to create it was using the `localVar` method.

Reply:
Ahhh.. yes, you are correct - my mistake!